pkg/scaffold/cmd.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaffold
+
+import (
+	"io"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+type cmd struct {
+	cmdInput *CmdInput
+}
+
+func NewCmdCodegen(in *CmdInput) Codegen {
+	return &cmd{cmdInput: in}
+}
+
+type CmdInput struct {
+	// ProjectPath is the project path rooted at GOPATH.
+	ProjectPath string
+}
+
+func (c *cmd) Render(w io.Writer) error {
+	o, err := yaml.Marshal(c.cmdInput)
+	if err != nil {
+		return err
+	}
+
+	_, err = w.Write(o)
```

Comment:
Why are we marshalling and writing out the struct into the writer? We're not even using the template here. 
I thought we need to execute the template based on the input struct:
```Go
t := template.New("main.go")

t, err := t.Parse(mainTmpl)
if err != nil {
	return err
}

return t.Execute(w, cm.CmdInput)
```

Reply:
you are right.

Reply:
fixed.